observer.go

```diff
@@ -0,0 +1,15 @@
+package opentracing
+
+// Observer can be registered with the Tracer to recieve notifications
```

Comment:
@yurishkuro that makes perfect sense; all I really care about is that the expectations are well-documented.

@hkshaw1990 I would feel a little more comfortable about this PR if there was some sort of clear disclaimer about the `Observer` being an alpha API that's subject to change.